sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/EncodingPerFile.java

```diff
@@ -44,10 +44,11 @@ boolean encodingMatch(InputFile inputFile) {
     }
 
     // Case-Insensitive check
-    Charset roslynEncoding = globalReportProcessor.getRoslynEncodingPerUri().get(uri); 
+    Charset roslynEncoding = globalReportProcessor.getRoslynEncodingPerUri().get(uri);
     if (roslynEncoding == null) {
-      LOG.warn("File '{}' does not have encoding information. Skip it.", uri);
-      return false;
+      // Roslyn saw the file, but it didn't recognized it's encoding. We fall back to the default encoding and accept the file.
+      LOG.warn("Roslyn encoding was not found for '{}', using default instead.", uri);
+      return true;
```

Comment:
Shouldn't this be an info message instead of warning? Since we populate the encoding and we don't provide any advice for the user, I'm not sure the user can do something about it.

Reply:
I'd prefer a warning. It should be rare, as I wasn't able to reproduce it. And it could lead to misencoded files in the SQ/SC if the situation happens.

And since the default isn't UTF-8 for some weird java-world reasons, user can either explicitly use the file with BOM, configure `<CodePage>` in project file or set `sonar.sourceEncoding` to fix this.